pkg/labels/oplabels.go

```diff
@@ -208,7 +208,13 @@ func (l Labels) upsertLabel(label Label) bool {
 		if label.Value == oldLabel.Value && label.Source == oldLabel.Source {
 			return false // No change
 		}
+
+		// If the label is not from the same source, then don't replace it.
```

Comment:
Shouldn't we pass in the `source` from `Replace{Identity,Info}Labels` and compare to that? I find it a bit weird that

- for upsert, we only compare sources to existing labels, but for anything new we don't consider the source
- for delete (aka keep), we consider the the source passed in the previous commit

In other words (assuming an empty endpoint that doesn't have any labels), I would expect that if you pass in `labels={cilium:foo=bar,nginx:baz=qux},source=nginx`  to `Replace{Identity,Info}Labels`, that it would only insert the `nginx` label, and ignore the "cilium:foo" label. 

But currently, if there is no `foo` label, it will also insert the `cilium:foo` label.

Is this behavior intentional? If so, why?

Reply:
> The issue was that if the "source" is defined in the label, having another parameter to receive the "source" would look odd?

To me, this doesn't seem particularly odd. `source` acts like a filter on labels, we could even rename it to `sourceFilter` to make it more clear. If you provide a source, then it becomes clear that anything within `labels` which does not match the `sourceFilter` will be ignored.

> Should I go with 2 and keep this current change?

The parameter would still exist for `Replace{Identity,Info}Labels`, right? In that case it doesn't make a difference to me, since my concern is more with the (imo surprising) behavior and less with the function signatures.

Reply:
> To me, this doesn't seem particularly odd. `source` acts like a filter on labels, we could even rename it to `sourceFilter` to make it more clear. If you provide a source, then it becomes clear that anything within `labels` which does not match the `sourceFilter` will be ignored.

That's the semantics I assumed during my review before I noticed you already commented here. I agree that renaming the parameter to `sourceFilter` and documenting it in the godoc would make its purpose clearer.